cartography/intel/aws/rds.py

```diff
@@ -200,17 +200,23 @@ def _attach_read_replicas(neo4j_session, read_replicas, aws_update_tag):
     Attach read replicas to their source instances
     """
     attach_replica_to_source = """
-    MATCH (replica:RDSInstance{id:{ReplicaArn}}),
-    (source:RDSInstance{db_instance_identifier:{SourceInstanceIdentifier}})
+    MATCH (replica:RDSInstance{id:{ReplicaArn}})
+    MERGE (source:RDSInstance{id:{SourceArn}})
     MERGE (replica)-[r:IS_READ_REPLICA_OF]->(source)
     ON CREATE SET r.firstseen = timestamp()
     SET r.lastupdated = {aws_update_tag}
     """
     for replica in read_replicas:
+        # NOTE: AWS returns DBInstanceIdentifier for intra-region replicas
+        # but returns ARN for inter-region replicas
+        source_arn = replica['ReadReplicaSourceDBInstanceIdentifier']
+        if not source_arn.startswith("arn:"):
+          arn_prefix = ":".join(replica['DBInstanceArn'].split(":")[:-1])
```

Comment:
Side note: I was about to suggest just using a string concatenation like this 
`arn:aws:rds:{region}:{account}:db:{instance_identifier}` to construct the arn, but apparently [other regions like China use `arn:aws-cn:..etc...`](https://docs.amazonaws.cn/en_us/AmazonRDS/latest/UserGuide/USER_Tagging.ARN.html), so your approach of lopping off the end and putting it back on is better.

Reply:
Lucky break for me - makes me look like I did something smart on purpose 😄 !